### PR TITLE
[AMDGPU] GFX12: Add Split Workgroup Barrier

### PR DESCRIPTION
**Pull request by mariusz-sikora-at-amd**

### CODE REVIEW

**Comment by mariusz-sikora-at-amd** [comment]
ping

File: llvm/test/MC/Disassembler/AMDGPU/gfx12_dasm_sop1.txt [hunk 1]
```diff
@@ -684,6 +684,59 @@
...
+0x7d,0x52,0x80,0xbe
+
+# GFX12: s_wakeup_barrier 1                       ; encoding: [0x81,0x57,0x80,0xbe]
+0x81,0x57,0x80,0xbe
+
+# GFX12: s_wakeup_barrier -1                      ; encoding: [0xc1,0x57,0x80,0xbe]
+0xc1,0x57,0x80,0xbe
+
+# GFX12: s_wakeup_barrier m0                      ; encoding: [0x7d,0x57,0x80,0xbe]
+0x7d,0x57,0x80,0xbe
```
Comment (piotrAMD) [inline llvm/test/MC/Disassembler/AMDGPU/gfx12_dasm_sop1.txt:730]: Nit: missing blank line between test cases.

File: llvm/test/MC/Disassembler/AMDGPU/gfx12_dasm_sopp.txt [hunk 2]
```diff
@@ -15,6 +15,15 @@
...
 
+# GFX12: s_barrier_wait 0xffff                   ; encoding: [0xff,0xff,0x94,0xbf]
+0xff,0xff,0x94,0xbf
+
+# GFX12: s_barrier_wait 1                        ; encoding: [0x01,0x00,0x94,0xbf]
+0x01,0x00,0x94,0xbf
+
+# GFX12: s_barrier_leave                         ; encoding: [0x00,0x00,0x95,0xbf]
+0x00,0x00,0x95,0xbf
+
```
Comment (piotrAMD) [inline llvm/test/MC/Disassembler/AMDGPU/gfx12_dasm_sopp.txt:26]: Nit: I think we are missing a single test for regular barrier here (s_barrier).

**Review by piotrAMD — APPROVED** [review]
LGTM with a nit.

File: llvm/test/MC/AMDGPU/gfx12_asm_sop1.s [hunk 3]
```diff
@@ -684,6 +684,51 @@ s_rndne_f16 s5, 0xfe0b
 s_rndne_f16 s5, 0x3456
 // GFX12: encoding: [0xff,0x6e,0x85,0xbe,0x56,0x34,0x00,0x00]
 
+s_barrier_signal -2
```
Comment (jayfoad) [inline llvm/test/MC/AMDGPU/gfx12_asm_sop1.s:687]: Missing `s_get_barrier_state` tests in this file?

Comment (mariusz-sikora-at-amd) [reply llvm/test/MC/AMDGPU/gfx12_asm_sop1.s:687]: Thanks !

Comment (mariusz-sikora-at-amd) [reply llvm/test/MC/AMDGPU/gfx12_asm_sop1.s:687]: Patch: https://github.com/llvm/llvm-project/pull/75575

**Pull request merged**